torch/nn/parallel/distributed.py

```diff
@@ -139,10 +138,16 @@ def __init__(self, module, device_ids=None, output_device=None, dim=0):
             if bucket_bytes >= bucket_bytes_cap:
                 self.bucket_sizes.append(0)
                 bucket_bytes = 0
-            self.bucket_sizes[-1] += 1
             for p in param_tuple:
-                self.bucket_map[p] = len(self.bucket_sizes) - 1
-            bucket_bytes += p.numel() * p.element_size()
+                add_to_bucket = False
```

Comment:
Wouldn't it be just simpler to do `if not param_tuple[0].requires_grad: continue` at the top of the loop?